### ISSUE TITLE
Markdown figures have different pixel dimensions on non-HiDPI monitors

### ISSUE THREAD

**Issue by Mashin6** [rstudio/rstudio#12679]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2023.03.0-daily+293
    OS Version      : MacOS 13.2
    R Version       : 4.2.1
    External monitor: ThinkVision P24H-10 QHD

This issue came up while testing fix for a related bug #7028.  When figure is generated on monitor with non-HiDPI resolution (e.g. 2560×1440 QHD) it has half pixels in each dimensions than figure generated on HiDPI resolution.

```
{r fig.width=4, fig.height=4}
plot(iris$Sepal.Length, iris$Sepal.Width)

```

retina display - all resolutions - figure dimensions 768 × 768 pixels
![retina-1512x982](https://user-images.githubusercontent.com/5265707/217197783-107ca161-871a-43ed-a505-f8a16a9ff1ec.png)

External Monitor - 2560x1140 (native) -  figure dimensions 384 × 384 pixels
![External-2560x1440](https://user-images.githubusercontent.com/5265707/217199707-d339bc69-698f-4c30-b4b9-7015c367b73e.png)

External Monitor - 1280x720 (HiDPI scale) -  figure dimensions 768 × 768 pixels
![External-1280x720](https://user-images.githubusercontent.com/5265707/217199996-96c3d7e7-0dda-489c-8a6e-8d5cbde5334a.png)

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@Mashin6 Thank you for opening a new issue for this! Do you by chance know whether or not this worked correctly in earlier versions of RStudio?

**Comment by Mashin6** [comment]
Previous and also current stable version of RStudio have the same issue + there is inconsistency in sizes of plot elements (reported in #7028). Fix introduced in latest daily build `2023.03.0-daily+293` solved the issue with inconsistent sizes of elements inside a plot, but still left the issue of different figure dimensions.

Here is also my previous report (RMarkdown repo) which shows the different pixel dimensions between figure on QHD and retina display. https://github.com/rstudio/rmarkdown/issues/1827#issuecomment-1009296406

**Comment by rdalbanus** [comment]
Hey folks,

I've struggled with this issue for over a year on our server's RStudio Server, even after upgrading to >2023.03.0*. 

It's the same thing as others described - I run a code chunk in my 4K monitor, and the plots are ~twice the expected size. Then I just drag the browser window to my adjacent 1440p monitor, rerun the chunk, and the plots are back to the expected size...

I don't have the issue on my local RStudio (RStudio 2024.04.2), so I'm wondering if it could be something specific to RStudio Server. I'm still trying to figure out how to approach this, so I'd appreciate any feedback. I'm happy to open a new issue, but I just wanted to touch base before.

Thanks!

\* Note: Our server is still on rhel 7, so we cannot upgrade to the newer 2024 RStudio releases. Nonetheless, this shouldn't be an issue after the 2023.03.0 release.

## System details
RStudio Edition : Server
RStudio Version              : 2023.12.1+402
OS Version                      : MacOS 14.4.1
R Version                         : 4.4.0
External monitor (4K)        : Dell P2723QE
External monitor (1440p) :  ASUS PA278CV

**RStudio Server** (with plot size issues):
RStudio 2023.12.1+402 "Ocean Storm" Release (4da58325ffcff29d157d9264087d4b1ab27f7204, 2024-01-29) for CentOS 7
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.4.1 Safari/605.1.15

**Local RStudio**:
RStudio 2024.04.2+764 "Chocolate Cosmos" Release (e4392fc9ddc21961fd1d0efd47484b43f07a4177, 2024-06-05) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2024.04.2+764 Chrome/120.0.6099.291 Electron/28.3.1 Safari/537.36, Quarto 1.4.555
